src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/DynamicEvents/DynamicEvents.cs

```diff
@@ -43,69 +43,85 @@ public sealed class DynamicEventSchema
 
         public int MaxOccurrence { get; init; } = 1;
 
-        public static void Set(List<DynamicEventSchema> dynamicEventSchemas, bool allowPartialSchema = true)
+        static DynamicEventSchema()
         {
-            DynamicEventSchemas.Clear();
-            DynamicEventSchema.allowPartialSchema = allowPartialSchema;
-            foreach (DynamicEventSchema dynamicEventSchema in dynamicEventSchemas)
+            // One time add.
+            if (DynamicEventSchemas.Count == 0)
+            {
+                // Add the adaptation schemas.
+                Add(GCDynamicEvents.SizeAdaptationSampleSchema);
+                Add(GCDynamicEvents.SizeAdaptationTuningSchema);
+            }
+        }
+
+        public static void Add(DynamicEventSchema dynamicEventSchema, bool allowPartialSchema = true)
+        {
+            if (DynamicEventSchemas.ContainsKey(dynamicEventSchema.DynamicEventName))
+            {
+                throw new Exception($"Provided schema has a duplicated event named {dynamicEventSchema.DynamicEventName}");
+            }
+            CompiledSchema schema = new CompiledSchema();
+            if (dynamicEventSchema.MinOccurrence < 0)
```

Comment:
Since you have an abstract type for `dynamicEventSchema`, it seems like it should be the place to restrict `MinOccurrence >= 0`.

Reply:
(and max >= min)